api.go

```diff
@@ -100,17 +97,38 @@ func (a *netAPI) doRequest(ctx context.Context, requestPath string, req proto.Me
 	if err != nil {
 		return err
 	}
-	return proto.Unmarshal(body, resp)
+	return jsonpb.UnmarshalString(string(body), resp)
```

Comment:
Would this throw an exception or similar if the input is malformed?

Reply:
Yes, and it would pass that through as the output.

Reply:
Can you add a test case for this? just to be sure

Reply:
Added.